packages/phpstan-rules/packages/cognitive-complexity/src/AstCognitiveComplexityAnalyzer.php

```diff
@@ -30,6 +33,16 @@ public function analyzeClassLike(ClassLike $classLike): int
             $totalCognitiveComplexity += $this->analyzeFunctionLike($classMethod);
         }
 
+        // non final classes are more complex
+        if (!$classLike->isFinal()) {
+            $totalCognitiveComplexity += self::NON_FINAL_CLASS_SCORE;
+        }
+
+        // classes extending from another are more complex
+        if ($classLike->extends !== null) {
+            $totalCognitiveComplexity += self::INHERITANCE_CLASS_SCORE;
+        }
+
```

Comment:
This should be located in separate service, to keep original complexity analyzer clean as possible. 

The extra service should not mix here, so we can easily maintain it and see original analyzer and the "extra" one.